### ISSUE TITLE
[Bug/Cleanup] Enforce shape contract for sigma in GaussianTaskLikelihood.predict

### ISSUE THREAD

**Issue by hmd101** [flatironinstitute/psyphy#142]
**Background**

`GaussianTaskLikelihood.loglik` vmaps `predict` over trials and passes the resulting `sigma` batch to `jsp.multivariate_normal.logpdf`. There is currently no enforced contract on what shape `predict` must return for sigma, which causes fragile reshape logic and inconsistent test mocks.

**Current workaround**

```
sigma = jnp.squeeze(sigma)
if n_trials == 1:
    sigma = jnp.expand_dims(sigma, axis=0)
```

This was introduced to handle the case where `predict` returns a `(r, r)` matrix per trial and vmap produces `(n_trials, r, r)`. When `r=1` and `n_trials=1` the batched shape is `(1, 1, 1)` — squeeze removes all dimensions including the trial axis, leaving a scalar with no axis for the subsequent vmap to map over.

**Problem: the fix doesn't generalize**

The squeeze + expand_dims only works correctly if `predict` returns sigma as a matrix. If `predict` returns a scalar (as `MockGaussianTask` in `test_likelihood_logic.py` currently does), `expand_dims` produces a `(1, 1)` vector for `n_trials=1`. The subsequent vmap then peels off the leading axis giving `s` shape `(1,)`  (a 1D vector),  which `multivariate_normal.logpdf` rejects because it expects a 2D matrix.

This failure is currently dormant: `test_likelihood_logic.py` uses `n_trials=2`, so the `n_trials=1` branch is never taken and the scalar path happens to work because `multivariate_normal.logpdf` accepts scalar cov for the 1D case. However, `n_trials=1` arises naturally in practice, for example during single-trial debugging, online inference, or leave-one-out cross-validation, and would trigger a silent shape error.

**Inconsistent test mocks**

The two test files already use different conventions for the 1D case:

```
# test_likelihood_logic.py — scalar sigma (wrong)
return (jnp.array(0), jnp.array(1))

# test_input_dim_validation.py ,  (1,1) matrix sigma (correct)
return (jnp.array(0.0), jnp.array([[1.0]]))
```

Neither file is clearly wrong because there is no documented or enforced contract. This means future implementors of concrete Gaussian tasks have no authoritative guidance and no early error if they get it wrong.

**Root cause**

`GaussianTaskLikelihood.predict` docstring says sigma should be an `(r x r)`  covariance matrix (r is the response dimension, ie how many values a single trial’s response consists of), but this is not enforced and the 1D case is ambiguous where both a scalar, `(1,)` vector and `(1,1)` matrix are all plausible interpretations.

**Suggested fix**

1. Document in the `GaussianTaskLikelihood` class docstring and `predict` docstring that sigma **must** always be returned as an `(r, r)` matrix — for `r=1` this means `jnp.array([[1.0]])`, never a scalar or vector
2. Remove the squeeze + expand_dims block → with the contract enforced, sigma after vmap is always `(n_trials, r, r)` and can be passed directly
3. Add a runtime shape check at the top of `loglik` to catch violations early:

```
mu, sigma = jax.vmap(...)(stimuli, trial_keys)

if sigma.ndim != 3 or sigma.shape[1] != sigma.shape[2]:
    raise ValueError(
        f"predict must return sigma as an (r, r) matrix per trial. "
        f"Got batched sigma shape {sigma.shape},  expected (n_trials, r, r). "
        f"For r=1 use jnp.array([[v]]), not a scalar."
    )
```

Important: This check runs before any further computation, and hence, works correctly under `jax.jit` (shapes are static at trace time), and gives a clear error pointing at the implementation mistake in `predict`.

1. Update `MockGaussianTask` in `test_likelihood_logic.py` to match the contract:

`return (jnp.array([0.0]), jnp.array([[1.0]]))`
